From ee0050ac21c85c0656f5ebd4c1cf3f69f76a9973 Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Fri, 11 Apr 2014 11:06:06 -0700
Subject: [PATCH] iio: mpu6050: Don't register irq handlers with trigger

Don't register local IRQ handlers with trigger IRQ. During buffer
disable from user space, chip is powered off by this driver. But
when trigger IRQ is freed then it will still try to access FIFOs
on powered off chip.

Change-Id: I064e1990d0d963f42ce107de2820bbc70d485668
Signed-off-by: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>
---
 drivers/iio/imu/inv_mpu6050/inv_mpu_core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c b/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c
index e062acd..b5db4ed 100644
--- a/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c
+++ b/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c
@@ -747,8 +747,8 @@ static int inv_mpu_probe(struct i2c_client *client,
 	indio_dev->modes = INDIO_BUFFER_TRIGGERED;
 
 	result = iio_triggered_buffer_setup(indio_dev,
-					    inv_mpu6050_irq_handler,
-					    inv_mpu6050_read_fifo,
+					    &iio_pollfunc_store_time,
+					    NULL,
 					    NULL);
 	if (result) {
 		dev_err(&st->client->dev, "configure buffer fail %d\n",
-- 
1.8.3.2

